lightning/src/ln/offers_tests.rs

```diff
@@ -1410,7 +1410,41 @@ fn creates_offer_with_blinded_path_using_unannounced_introduction_node() {
 	route_bolt12_payment(bob, &[alice], &invoice);
 	expect_recent_payment!(bob, RecentPaymentDetails::Pending, payment_id);
 
-	claim_bolt12_payment(bob, &[alice], payment_context, &invoice);
+	// Extract PaymentClaimable, verify context, and obtain the payment preimage.
+	let recipient = &alice;
+	let payment_purpose = match get_event!(recipient, Event::PaymentClaimable) {
+		Event::PaymentClaimable { purpose, .. } => purpose,
+		_ => panic!("No Event::PaymentClaimable"),
+	};
+
+	let payment_preimage = payment_purpose
+		.preimage()
+		.expect("No preimage in Event::PaymentClaimable");
+
+	match payment_purpose {
+		PaymentPurpose::Bolt12OfferPayment { payment_context: ctx, .. } => {
+			assert_eq!(PaymentContext::Bolt12Offer(ctx), payment_context);
+		},
+		PaymentPurpose::Bolt12RefundPayment { payment_context: ctx, .. } => {
+			assert_eq!(PaymentContext::Bolt12Refund(ctx), payment_context);
+		},
+		_ => panic!("Unexpected payment purpose: {:?}", payment_purpose),
+	};
```

Comment:
Could this be DRY'ed up by making a helper used in `claim_bolt12_payment` that can also be used here?

Reply:
I introduced another helper, to DRY up the code a little bit in [.08](https://github.com/shaavan/rust-lightning/commits/pr4152.08). Let me know if it looks good!